### PR TITLE
use setState instead of forceUpdate in main crossword component

### PR DESCRIPTION
**Pull request by desbo**
cc @rich-nguyen @OliverJAsh

### CODE REVIEW

**Comment by OliverJAsh** [comment]
@robertberry Do you have any thoughts on this?

File: static/src/javascripts/es6/projects/common/modules/crosswords/main.js [hunk 1]
```diff
@@ -317,36 +324,56 @@ class Crossword extends React.Component {
...
+            });
 
             setTimeout(() => {
-                _.forEach(badCells, (cell) => {
-                    cell.isError = false;
-                    cell.value = '';
+                this.setState({
+                    grid: helpers.mapGrid(this.state.grid, (cell, gridX, gridY) => {
+                        if (_.some(badCells, bad => bad.x === gridX && bad.y === gridY)) {
+                            cell.isError = false;
```
Comment (OliverJAsh) [inline static/src/javascripts/es6/projects/common/modules/crosswords/main.js]: Is `cell` a new object, or are we mutating the old copy? I'm not sure about the implementation of `helpers.mapGrid`.

**Comment by OliverJAsh** [comment]
:+1:. So, as far as I understand, because we're creating a new reference entirely, it doesn't need to do a deep equality check?

**Comment by desbo** [comment]
@OliverJAsh yeah, instead of mutating `this.state.grid` in-place we build a totally new grid and use `setState` to replace the old grid with the new one. this way there's no direct mutation of `this.state` and no need to use `forceUpdate`, which is preferable IMO.

**Comment by rich-nguyen** [comment]
Agreed, @desbo

File: static/src/javascripts/es6/projects/common/modules/crosswords/main.js (hunk 1 above)
Comment (desbo) [inline static/src/javascripts/es6/projects/common/modules/crosswords/main.js]: we're mutating the cells. `mapGrid` gives you a new grid with a distinct reference but the cells are the same objects as before.

**Comment by OliverJAsh** [comment]
:+1: :+1:

**Pull request merged**